docs/design/drf - fairshare.md

```diff
@@ -0,0 +1,179 @@
+# Namespace fair share
+
+[@lminzhw](http://github.com/lminzhw); May 8, 2019
+
+## Motivation
+
+`Queue` was introduced in [kube-batch](http://github.com/kubernetes-sigs/kube-batch) to share resources among users.
+
+But, the user in the same `Queue` are equivalent during scheduling. For example, we have a `Queue` contains a small amount of resources, and there are 10 pods belong to UserA and 1000 pods belong to UserB. In this case, pods of UserA would have less probability to bind with node.
+
+So, we need a more fine-grained strategy to balance resource usage among users in the same `Queue`.
+
+In consideration of multi-user model in kubernetes, we use namespace to distinguish different user. Each namespace would have its weight to control resources usage.
+
+## Function Specification
+
+Weight have these features:
+> 1. `Queue` level
+> 2. an `integer` with default value 1
+> 3. record in namespace `quota`
+> 4. higher value means more resources after balancing
+
+### where is the weight
+
+```yaml
+apiVersion: v1
+kind: ResourceQuota
+metadata:
+  namespace: default
```

Comment:
Is `ResourceQuota` same as Kubernetes [`ResourceQuota`](https://kubernetes.io/docs/concepts/policy/resource-quotas/)? If not, could `volcano.sh/namespace.weight` be named to `weight` because the whole type is `volcano.sh` related? 

Does it need a `name`? Then we could create different quotas of same namespace for different queues.

Reply:
it's Kubernetes `ResourceQuota` as far as I known :)

Reply:
If it is same, it is a little inappropriate to put `volcano.sh/namespace.weight` in `.spec.hard`, because `hard is the set of desired hard limits for each named resource` and `weight` is not a hard limit.

Reply:
> hard is the set of desired hard limits for each named resource and weight is not a hard limit.

It's a kind of `hard` as scheduler will make sure namespace will not be overused.

Reply:
It is OK if it is a hard limit.

Reply:
> It is OK if it is a hard limit.

And once the resource reached the limit, there will be no more pods created for the namespace.